### PR TITLE
[devel_transport] Port tests to new version of greentea

### PR DESCRIPTION
**Pull request by stegut01**
Test results from ```$mbedgt -VS --report-junit JUnit.xml```:
```
mbedgt: test suite report:
+---------------+---------------+-----------------------------+--------+--------------------+-------------+
| target        | platform_name | test suite                  | result | elapsed_time (sec) | copy_method |
+---------------+---------------+-----------------------------+--------+--------------------+-------------+
| frdm-k64f-gcc | K64F          | minar-test-complex_dispatch | OK     | 39.77              | shell       |
| frdm-k64f-gcc | K64F          | minar-test-dispatchtest     | OK     | 24.93              | shell       |
+---------------+---------------+-----------------------------+--------+--------------------+-------------+
mbedgt: test suite results: 2 OK
mbedgt: test case report:
+---------------+---------------+-----------------------------+------------------+--------+--------+--------+--------------------+
| target        | platform_name | test suite                  | test case        | passed | failed | result | elapsed_time (sec) |
+---------------+---------------+-----------------------------+------------------+--------+--------+--------+--------------------+
| frdm-k64f-gcc | K64F          | minar-test-complex_dispatch | complex_dispatch | 1      | 0      | OK     | 39.77              |
| frdm-k64f-gcc | K64F          | minar-test-dispatchtest     | dispatchtest     | 1      | 0      | OK     | 24.93              |
+---------------+---------------+-----------------------------+------------------+--------+--------+--------+--------------------+
mbedgt: test case results: 2 OK
mbedgt: completed in 66.22 sec
```
```
$yt ls
minar 1.0.4
|_ compiler-polyfill 1.2.1
|_ minar-platform 1.0.0
| |_ minar-platform-mbed 1.1.2 yotta_modules\minar-platform-mbed
| | \_ mbed-hal 1.2.2 yotta_modules\mbed-hal
| |   |_ mbed-drivers 1.0.0 yotta_modules\mbed-drivers
| |   | \_ ualloc 1.0.3 yotta_modules\ualloc
| |   |   \_ dlmalloc 1.0.0 yotta_modules\dlmalloc
| |   \_ mbed-hal-freescale 1.0.0 yotta_modules\mbed-hal-freescale
| |     \_ mbed-hal-ksdk-mcu 1.0.8 yotta_modules\mbed-hal-ksdk-mcu
| |       |_ uvisor-lib 1.0.13 yotta_modules\uvisor-lib
| |       \_ mbed-hal-k64f 1.1.0 yotta_modules\mbed-hal-k64f
| |         \_ mbed-hal-frdm-k64f 1.0.2 yotta_modules\mbed-hal-frdm-k64f
| \_ cmsis-core 1.1.2 yotta_modules\cmsis-core
|   \_ cmsis-core-freescale 1.0.0 yotta_modules\cmsis-core-freescale
|     \_ cmsis-core-k64f 1.0.0 yotta_modules\cmsis-core-k64f
|_ core-util 1.5.1
\_ greentea-client 0.1.4 (test dependency)
```

### CODE REVIEW

File: test/complex_dispatch.cpp [hunk 1]
```diff
@@ -113,6 +110,6 @@ void app_start(int, char*[]) {
     if (cb_cnt != EXPECTED_CALLBACK_COUNT)
```
Comment (mazimkhan) [inline test/complex_dispatch.cpp]: Looks like an error condition. Should we have following here:

```
TEST_ASSERT_EQUAL_MESSAGE(EXPECTED_CALLBACK_COUNT, cb_cnt, "Wrong call back count!");
```

**Comment by PrzemekWirkus** [comment]
+1

File: test/complex_dispatch.cpp [hunk 2]
```diff
@@ -113,6 +110,6 @@ void app_start(int, char*[]) {
...
         printf("ERROR: final callback count is %d, should be %d\r\n", cb_cnt, EXPECTED_CALLBACK_COUNT);
 
-    MBED_HOSTTEST_RESULT(cnt_ok && (cb_cnt == 1));
+    GREENTEA_TESTSUITE_RESULT(cnt_ok && (cb_cnt == 1));
```
Comment (mazimkhan) [inline test/complex_dispatch.cpp]: It should be ```EXPECTED_CALLBACK_COUNT ``` instead or ```1```

**Comment by LiyouZhou** [comment]
LGTM

**Comment by bogdanm** [comment]
LGTM

**Pull request merged**